staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -328,10 +328,15 @@ func (s *store) GuaranteedUpdate(
 			// etcd in order to be sure the data in the store is equivalent to
 			// our desired serialization
 			if mustCheckData {
-				origState, err = getCurrentState()
+				neworigState, err := getCurrentState()
 				if err != nil {
 					return err
 				}
+				// we are already up to date
+				if bytes.Equal(origState.data, neworigState.data) {
```

Comment:
Flipping this removes duplicated code, makes the loop restart case the exception, and removes the need for neworigState

```
if !bytes.Equal(data, origState.data) { 
  // the data changed, restart the loop
  continue
}
```

Reply:
Done.